gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java

```diff
@@ -134,10 +158,14 @@ public void When_MappingMedicationRequestWithAvailabilityTimeNotInMedicationStat
         var ehrExtract = unmarshallEhrExtract("ehrExtract_AvailabilityTimeNotInMedicationStatement.xml");
         var expectedAuthoredOn = DateFormatUtil.parseToDateTimeType("20100117");
 
-        when(medicationRequestPlanMapper.mapToPlanMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
-        when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
-        when(medicationStatementMapper.mapToMedicationStatement(any(), any(), any(), any(), any())).thenReturn(new MedicationStatement());
-        when(medicationMapper.createMedication(any())).thenReturn(new Medication());
+        when(medicationRequestPlanMapper.mapToPlanMedicationRequest(any(), any(), any(), any(), any()))
+            .thenReturn(new MedicationRequest());
+        when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(), any(), any(), any(), any()))
```

Comment:
I think here we can narrow it down to the classes that we are expecting to receive while mocking, something like below:
when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(RCMRMT030101UKEhrExtract.class),
                                                                      any(RCMRMT030101UKEhrComposition.class),
                                                                      any(RCMRMT030101UKMedicationStatement.class),
                                                                      any(RCMRMT030101UKPrescribe.class),
                                                                      anyString()))
            .thenReturn(new MedicationRequest());

Reply:
similarly in other places...